src/mrnet/network/reaction_network.py

```diff
@@ -1594,6 +1632,162 @@ def mols_w_cuttoff(RN_pr_solved, cutoff=0, build_pruned_network=True):
+            if len(combined_reactants) == 2 and len(combined_products) == 2:
+                node_str = str(combined_reactants[0]) + "+" + "PR_" + str(combined_reactants[
+                                                                              1]) + "," + str(
+                    combined_products[0]) + "+" + str(
+                    combined_products[1])
+            elif len(combined_reactants) == 2 and len(combined_products) == 1:
+                node_str = str(combined_reactants[0]) + "+" + "PR_" + str(combined_reactants[
+                                                                        1]) + "," + str(
+                    combined_products[0])
+            elif len(combined_reactants) == 1 and len(combined_products) == 2:
+                node_str = str(combined_reactants[0]) + "," + str(
+                    combined_products[0]) + "+" + str(
+                    combined_products[1])
+            elif len(combined_reactants) == 1 and len(combined_products) == 1:
+                node_str = str(combined_reactants[0]) + "," + str(
+                    combined_products[0])
+            else:
+                node_str = None
+                print("Something is wrong", combined_reactants, combined_products)
+        return node_str
+
+
+    @staticmethod
+    def parse_reaction_node(node):
+        """
+        Method to parse out reactant,pr and prodcut nodes from a node string.
+        :param node: node string, ex "1+PR_2,3+4"
+        """
+        react_list = []
+        prod_list = []
+        if "PR" in node and "+" in node.split(",")[1]:
+            reactant = int(node.split("+PR_")[0])
+            pr = int(node.split("+PR_")[1].split(",")[0])
+            product1 = int(node.split("+PR_")[1].split(",")[1].split("+")[0])
+            product2 = int(node.split("+PR_")[1].split(",")[1].split("+")[1])
+            react_list = [reactant, pr]
+            prod_list = [product1, product2]
+            prod_list.sort()
+        elif "PR" in node and "+" not in node.split(",")[1]:
+            reactant = int(node.split("+PR_")[0])
+            pr = int(node.split("+PR_")[1].split(",")[0])
+            product = int(node.split("+PR_")[1].split(",")[1])
+            react_list = [reactant, pr]
+            prod_list = [product]
+        elif "PR" not in node and "+" in node:
+            reactant = int(node.split(",")[0])
+            product1 = int(node.split(",")[1].split("+")[0])
+            product2 = int(node.split(",")[1].split("+")[1])
+            prod_list = [product1, product2]
+            prod_list.sort()
+            react_list = [reactant]
+            prod_list = [product1, product2]
+            prod_list.sort()
+        elif "PR" not in node and "+" not in node:
+            reactant = int(node.split(",")[0])
+            product = int(node.split(",")[1])
+            react_list = [reactant]
+            prod_list = [product]
+        else:
```

Comment:
You can condense this code dramatically. For instance, you don't actually need to check if "+" is in the second half of the string. You can just do something like `prod_list = [int(i) for i in node.split(",")[1].split("+")]`